heudiconv/bids.py

```diff
@@ -487,12 +486,34 @@ def convert_sid_bids(subject_id):
     subject_id : string
         Original subject ID
     """
+    sid, subject_id = sanitize_label(subject_id)
+    lgr.warning('Deprecation warning: convert_sid_bids() is deprecated, '
+                'please use sanitize_label() instead.')
+    
+    return sid, subject_id
+
+
+def sanitize_label(label):
+    """Strips any non-BIDS compliant characters within label
+
+    Parameters
+    ----------
+    label : string
+
+    Returns
+    -------
+    clean_label : string
+        New, sanitized label
+    label : string
+        Original label
+    """
     cleaner = lambda y: ''.join([x for x in y if x.isalnum()])
-    sid = cleaner(subject_id)
-    if not sid:
+    clean_label = cleaner(label)
+    if not clean_label:
         raise ValueError(
-            "Subject ID became empty after cleanup.  Please provide manually "
-            "a suitable alphanumeric subject ID")
-    lgr.warning('{0} contained nonalphanumeric character(s), subject '
-                'ID was cleaned to be {1}'.format(subject_id, sid))
-    return sid, subject_id
+            "Label became empty after cleanup.  Please provide manually "
+            "a suitable alphanumeric label.")
+    if clean_label != label:
+        lgr.warning('{0} contained nonalphanumeric character(s), label '
+                    'was cleaned to be {1}'.format(label, clean_label))
```

Comment:
```suggestion
        lgr.warning('%s contained nonalphanumeric character(s), label '
                    'was cleaned to be %s', label, clean_label)
```
so that if label or clean_label contains some %, logger doesn't freak out when tries to %-interpolate it

Reply:
I see that similar wrong code pattern was used before, but we better fix that since we found it now ;)